pkg/libvirttools/virtualization.go

```diff
@@ -1185,12 +1174,14 @@ func (v *VirtualizationTool) updateDomainMemory(domain virt.Domain, lcr *kubeapi
 		return fmt.Errorf("requested memory exceeds Max Memory configuration for the VM")
 	}
 
-	if newMemory != int64(currentMemory) {
-		glog.V(4).Infof("Update domain memory in KiB: %v -> %v", currentMemory, newMemory)
-		err := domain.AdjustDomainMemory(newMemory - currentMemory)
-		if err != nil {
-			return err
-		}
+	if newMemory == currentMemory {
+		return fmt.Errorf("requested memory is the same as current configured memoryß")
```

Comment:
It’s probably better to treat it as a success if the newMemory is equal to the currentMemory.

Reply:
yes, we should actually. i found this with the cases where only cpu is changed, the state can be setup a bit earlier than the cpu update at the libvirt domain. will make change.

Reply:
fixed. this case is blocked in the updateContainerResource() CRI interface and only call the updateMem or updateCpu as needed.